agent/src/main/java/io/lumigo/agent/Agent.java

```diff
@@ -11,19 +11,39 @@
 import java.nio.file.Paths;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.jar.JarFile;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class Agent {
 
+    private static boolean initialized = false;
+
+    public static String LUMIGO_JAVA_TRACER_PATH = "/opt/lumigo/lumigo-tracer.jar";
+
+    public static void premain(String agentArgs, Instrumentation inst) {
+        if (!isKillSwitchOn()) {
```

Comment:
I needed to add the kill switch also here because until now the tracer activate the agent and now the agent can be auto-activated with auto-trace